frontend/src/util/ZenonClient.ts

```diff
@@ -85,3 +88,135 @@ export async function getRecord(zenonID: string): Promise<ZenonRecord> {
+    return array.indexOf(value) === index;
+}
+
+function extractAuthors(record: ZenonRecord): Person[] {
+    return record.authors.map((author: Author) => {
+        const authorSplit = author.name.split(',');
+        if (authorSplit.length === 2) {
+            return {
+                givenname: authorSplit[1].replace(/[\\.]+$/, '').trim(),
+                lastname: authorSplit[0].trim()
+            } as Person;
+        }
+        return {
+            givenname: '',
+            lastname: author.name
+        } as Person;
+    });
+}
+export async function loadJournalZenonData(target: JobTargetData) {
+    try {
+        const zenonRecord = (await getRecord(
+            target.metadata.zenon_id
+        )) as ZenonRecord;
+        const errors: string[] = [];
+
+        let parentId = '';
+        if (!zenonRecord.parentId) {
+            errors.push(
+                `Zenon record has no parent id. Can't determine which Journal this issue belongs to.`
+            );
+        } else if (!(zenonRecord.parentId in ojsZenonMapping)) {
+            errors.push(
+                `Missing OJS Journal code for Journal with Zenon-ID '${zenonRecord.parentId}'.`
+            );
+        } else {
+            // eslint-disable-next-line prefer-destructuring
+            parentId = zenonRecord.parentId;
+        }
+
+        if (errors.length !== 0) {
+            return new JobTargetError(target.id, target.path, errors);
+        }
+
+        const metadata = {
+            zenon_id: target.metadata.zenon_id,
+            volume: getSerialVolume(zenonRecord),
+            publishing_year: parseInt(zenonRecord.publicationDates[0], 10),
+            number: getIssueNumber(zenonRecord),
+            description: zenonRecord.title,
+            ojs_journal_code: ojsZenonMapping[parentId],
+            reporting_year: getReportingYear(zenonRecord)
+        } as JournalIssueMetadata;
+
+        return new JobTargetData(target.id, target.path, metadata);
+    } catch (error) {
+        return new JobTargetError(target.id, target.path, [error]);
+    }
+}
+
+function getReportingYear(record: ZenonRecord): number {
+    const match = record.partOrSectionInfo.match(/\(.*?\)/g)![0];
+    return parseInt(match.slice(1, -1), 10);
+}
+
+function getIssueNumber(record: ZenonRecord): number {
+    if (!record.partOrSectionInfo.includes(',')) {
+        return 0;
+    }
+    const match = record.partOrSectionInfo.match(/,[^(]*/g)![0];
+    return parseInt(match.slice(1), 10);
+}
+
+function getSerialVolume(record: ZenonRecord): number {
+    let match: string = '';
+    if (record.partOrSectionInfo.includes(',')) {
+        // eslint-disable-next-line prefer-destructuring
+        match = record.partOrSectionInfo.match(/[^,]*/g)![0];
+    } else {
+        // eslint-disable-next-line prefer-destructuring
+        match = record.partOrSectionInfo.match(/[^(]*/g)![0];
+    }
+    return parseInt(match, 10);
+}
```

Comment:
Jetzt hast du aber quasi die Logik von Jobs in den ZenonClient geholt, was im Grunde das Problem der Kopplung nur verschiebt. 

Die Idee beim Ticket war ja, dass man das `ZenonRecord` entsprechend erweitert und dort ggf. sinnvolle neue Member-Variablen einfügt, auf die die Jobs dann einfach zugreifen.